libretro/libretro_core_options.h

```diff
@@ -0,0 +1,1359 @@
+   { "5", NULL },          \
+   { "6", NULL },          \
+   { "7", NULL },          \
+   { "8", NULL },          \
+   { "9", NULL },          \
+   { NULL, NULL },         \
+}
+
+/*
+ ********************************
+ * VERSION: 2.0
+ ********************************
+ *
+ * - 2.0: Add support for core options v2 interface
+ * - 1.3: Move translations to libretro_core_options_intl.h
+ *        - libretro_core_options_intl.h includes BOM and utf-8
+ *          fix for MSVC 2010-2013
+ *        - Added HAVE_NO_LANGEXTRA flag to disable translations
+ *          on platforms/compilers without BOM support
+ * - 1.2: Use core options v1 interface when
+ *        RETRO_ENVIRONMENT_GET_CORE_OPTIONS_VERSION is >= 1
+ *        (previously required RETRO_ENVIRONMENT_GET_CORE_OPTIONS_VERSION == 1)
+ * - 1.1: Support generation of core options v0 retro_core_option_value
+ *        arrays containing options with a single value
+ * - 1.0: First commit
+*/
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/*
+ ********************************
+ * Core Option Definitions
+ ********************************
+*/
+
+/* RETRO_LANGUAGE_ENGLISH */
+
+/* Default language:
+ * - All other languages must include the same keys and values
+ * - Will be used as a fallback in the event that frontend language
+ *   is not available
+ * - Will be used as a fallback for any missing entries in
+ *   frontend language definition */
+
+struct retro_core_option_v2_category option_cats_us[] = {
+   {
+      "system",
+      "System",
+      "Configure system options."
+   },
+   {
+      "video",
+      "Video",
+      "Configure video options."
+   },
+   {
+      "network",
+      "Network",
+      "Configure network options."
+   },
+   { NULL, NULL, NULL },
+};
+
+struct retro_core_option_v2_definition option_defs_us[] = {
+   {
+      "ppsspp_language",
+      "Language",
+      NULL,
+      "Choose language, 'Automatic' will use the frontend language.",
+      NULL,
+      "system",
+      {
+         { "Automatic",           NULL },
+         { "English",             NULL },
+         { "Japanese",            NULL },
+         { "French",              NULL },
+         { "Spanish",             NULL },
+         { "German",              NULL },
+         { "Italian",             NULL },
+         { "Dutch",               NULL },
+         { "Portuguese",          NULL },
+         { "Russian",             NULL },
+         { "Korean",              NULL },
+         { "Chinese Traditional", NULL },
+         { "Chinese Simplified",  NULL },
+         { NULL, NULL },
+      },
+      "Automatic"
+   },
+   {
+      "ppsspp_cpu_core",
+      "CPU Core",
+      NULL,
+      NULL,
+      NULL,
+      "system",
+      {
+         { "JIT",         "Dynarec (JIT)" },
+         { "IR JIT",      "IR Interpreter" },
+         { "Interpreter", NULL },
+         { NULL, NULL },
+      },
+      "JIT"
+   },
+   {
+      "ppsspp_fast_memory",
+      "Fast Memory (Speedhack)",
```

Comment:
Fast memory isn't a speedhack, it's something of the reverse - having it off is a "slowhack."  The actual PSP always has fast memory enabled.  However, if a bug occurs with fast memory enabled, you get a crash.  Some games (especially with adhoc, etc.) still run into emulator bugs, but if you force the game not to crash... it still mostly works.

That's what this setting does - when disabled, it forces the game not to crash when a PSP playing the game would've crashed (i.e. exiting to the XMB or powering itself off.)

I realize you seem to be just copying what some libretro person decided to call it from the other file.

-[Unknown]

Reply:
Thank you for the explanation! And yes I indeed copied most of the options as-is, "(Speedhack)" removed :)